### PR TITLE
Enable nullability in ToolStripPanelCell

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in ToolStripPanelCell.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8323)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelCell.cs [hunk 1]
```diff
@@ -298,41 +298,44 @@ protected override void SetBoundsCore(Rectangle bounds, BoundsSpecified specifie
             CachedBounds = bounds;
             try
             {
-                if (DraggedControl.IsCurrentlyDragging)
+                if (ToolStripPanelRow is not null)
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelCell.cs:301]: Can this be inverted?

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelCell.cs [hunk 2]
```diff
@@ -383,14 +386,17 @@ private void OnToolStripLocationChanging(object sender, ToolStripLocationCancelE
                     Point newloc = e.NewLocation;
                     // detect if we haven't yet performed a layout - force one so we can
                     // properly join to the row.
-                    if (ToolStripPanelRow is not null && ToolStripPanelRow.Bounds == Rectangle.Empty)
+                    if (ToolStripPanelRow is not null)
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelCell.cs:389]: Same here with inverting

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelCell.cs [hunk 3]
```diff
@@ -383,14 +386,17 @@ private void OnToolStripLocationChanging(object sender, ToolStripLocationCancelE
...
-                    }
+                        if (ToolStripPanelRow.Bounds == Rectangle.Empty)
+                        {
+                            ToolStripPanelRow.ToolStripPanel.PerformUpdate(true);
+                        }
 
-                    if (_wrappedToolStrip is not null)
-                    {
-                        ToolStripPanelRow.ToolStripPanel.Join(_wrappedToolStrip, newloc);
+                        if (_wrappedToolStrip is not null)
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelCell.cs:396]: Can `_wrappedToolStrip` be `null`? If so, should it also be null annotated at the top where we declare the fields?

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelCell.cs [hunk 4]
```diff
@@ -34,11 +32,12 @@ internal class ToolStripPanelCell : ArrangedElement
         private static int t_cellCount;
 #endif
 
-        public ToolStripPanelCell(Control control) : this(null, control)
+        public ToolStripPanelCell(Control control)
+            : this(null, control)
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripPanelCell.cs:36]: nit: argument name for null/constant parameters.

**Review by lonitra — APPROVED** [review]
🚀

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**